src/modal.tsx

```diff
@@ -353,14 +353,16 @@ export class ReactNativeModal extends React.Component<ModalProps, State> {
             /scrollview|flatlist/i.test(instance.type),
           );
 
-        if (
-          hasScrollableView &&
-          this.shouldPropagateSwipe(e, gestureState) &&
-          this.props.scrollTo &&
-          this.props.scrollOffset > 0
-        ) {
+        // when users scroll over inner scrollview/flatlist the modal shouldn't responses
+        // if propagateSwipe is set
+        if (hasScrollableView && this.shouldPropagateSwipe(e, gestureState)) {
```

Comment:
if I remember correctly, I gave up on this solution, so the case did not work correctly when the modal is in the scrollView.
In this case, hasScrollableView is always true. although there is no scrollView inside the modal.
In addition to this, I think you should rewrite the logic for hasScrollableView. So searches  /scrollview|flatlist/ should be only in inside of modal.

Try checking something like this.
```
<ScrollView> 
  <Modal
      ...
      propagateSwipe={true}
  >
   <View>
     <Text> Hello </Text>
   </View>
  </Modal>
</ScrollView>
```

Reply:
Can confirm :/